docs/sources/userguide/plugins.md

```diff
@@ -0,0 +1,157 @@
+Docker Plugins MVP
+==================
+
+Plugins provide a mechanism for hooking into the Docker engine to extend its behavior.
+
+HTTP over a local UNIX socket is used to communicate between Docker and its plugins.
+
+Plugins are distributed as containers.
```

Comment:
My take on these:

- I'd rather *not* expose `docker run` arguments to `docker plugin` for two reasons
   - As soon as we start adding some of them, we'll end up adding all of them.
   - I like that we have a dedicated `docker plugin` top-level command that hides the fact that it's actually pulling an image and running a container: I believe makes perfect sense for an MVP, but I don't think we'd like to lock ourselves in that as the only possible execution model.
- Mounting the daemon socket should not be the default, although I agree it should be made possible through some command-line flag that makes it obvious you're doing something "privileged", such as `--allow-full-daemon-access`.